api/leaderworkerset/v1/leaderworkerset_types.go

```diff
@@ -133,6 +152,17 @@ type RolloutStrategy struct {
 	RollingUpdateConfiguration *RollingUpdateConfiguration `json:"rollingUpdateConfiguration,omitempty"`
 }
 
+// SubGroupingPolicy describes the policy that will be applied when creating subgroups.
+type SubGroupingPolicy struct {
+	// The number of pods per subgroup. This value is immutable,
+	// and must not be greater than LeaderWorkerSet.Spec.Size.
+	// Size must be divisible by subGroupSize in which case the
+	// subgroups will be of equal size. Or size - 1 is divisible
+	// by subGroupSize, in which case the leader is considered as
+	// the extra pod, and will be part of the first subgroup.
+	SubGroupSize *int32 `json:"subGroupSize,omitempty"`
```

Comment:
Is there any validation logic for this field? we should have the validation logic 1. The value is smaller than lws.size and size/subgroupsize or (size-1)/subgroupsize is 0; 2. When the subgroup exclusive placement annotation is added, the subgroupsize field is added as well 3 we don't allow updating the field for SubGroupPolicy once exists